shared/src/main/scala/amf/client/remod/AMFClient.scala

```diff
@@ -11,29 +11,29 @@ import amf.core.resolution.pipelines.ResolutionPipeline
 import amf.core.services.RuntimeCompiler
 import amf.core.validation.AMFValidationReport
 
-import scala.concurrent.Future
-
+import scala.concurrent.{ExecutionContext, Future}
 
 // client
-private[remod] class AMFClient(env: BaseEnvironment) {
+private[amf] class AMFClient(protected val configuration: AMFConfiguration) {
 
+  implicit val exec: ExecutionContext = configuration.getExecutionContext
 
-  def getEnvironment:BaseEnvironment = env
+  def getEnvironment: AMFConfiguration = configuration
   // how do we return the ErrorHandler that is created by the provider?
   // relation between EH vs BU? should the BU have an EH inside?
   // should parse always return error handler + base unit? (Amf result)
 
   // sync or async?
 
   // content type format, pendiente
-  def parse(url: String, vendor: Option[Vendor] = None): Future[AmfResult] = AMFParser.parse(url, env)
-    // build parsing context?
+  def parse(url: String, vendor: Option[Vendor] = None): Future[AMFResult] = AMFParser.parse(url, configuration)
+  // build parsing context?
 
-  def resolve(bu: BaseUnit): AmfResult = AMFResolver.resolve(bu, env) // clone? BaseUnit.resolved
+  def resolve(bu: BaseUnit): AMFResult = AMFResolver.resolve(bu, configuration) // clone? BaseUnit.resolved
 
-  def render(bu:BaseUnit, target:Vendor):String = AMFRender.render(bu, env)
+  def render(bu: BaseUnit, target: Vendor): String = AMFRenderer.render(bu, configuration)
```

Comment:
Same here with target (Vendor). The AMFRenderer method didn't receive it, but there is one that receive it as a String... I think it should be typed as Vendor

Reply:
We are having this discussion on the renderer class. Closing this conversation to centralize the subject in just one place.